controllers/databasecluster_controller.go

```diff
@@ -607,11 +655,11 @@ func (r *DatabaseClusterReconciler) getSecretFromMonitoringConfig(
 	return secretData, nil
 }
 
-// createOrUpdateSecret creates or updates a secret by its name.
+// CreateOrUpdateSecret creates or updates a secret by its name.
 // If a secret is created and secretName is empty, generateName is used
 // to generate a unique name.
-func (r *DatabaseClusterReconciler) createOrUpdateSecret(
-	ctx context.Context, database *everestv1alpha1.DatabaseCluster,
+func CreateOrUpdateSecret(
```

Comment:
I suggest we extract the methods which are no longer in `DatabaseClusterReconciler` into a separate file so it's clear they can be reused.

Reply:
I also thought about that and agree that it's the best approach going forward. I'll make that change.